hardhat/test/mineToken/delegation.test.ts

```diff
@@ -0,0 +1,237 @@
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers'
+import { deployMineFixture } from '../fixtures/deployMineFixture'
+import { expect } from 'chai'
+import { randomBytes } from 'ethers'
+import { ethers } from 'hardhat'
+
+describe('MineToken delegations', () => {
+  const DelegationType = {
+    Delegation: [
+      { name: 'delegatee', type: 'address' },
+      { name: 'nonce', type: 'uint256' },
+      { name: 'expiry', type: 'uint256' },
+    ],
+  }
+  it('should delegate vote', async () => {
+    const { mine, other, owner } = await loadFixture(deployMineFixture)
+    await mine.delegate(other.address)
+    expect(await mine.delegatees(owner.address)).to.equal(other.address)
+  })
+
+  it('mint should set default delegatee if no delegatee before', async () => {
+    const { mine, other } = await loadFixture(deployMineFixture)
+    await mine.mint(other.address, BigInt(100000) * BigInt(10) ** BigInt(18))
+    const defaultDelegatee = await mine.defaultDelegatee()
+    expect(await mine.delegatees(other.address)).to.equal(defaultDelegatee)
+    const expectedVotesOfDefaultDelegatee =
+      (await mine.balanceOf(other.address)) + (await mine.balanceOf(other.address))
```

Comment:
I suppose we should be adding `owner` votes and `other` votes to get all votes of default delegatee
```suggestion
      (await mine.balanceOf(owner.address)) + (await mine.balanceOf(other.address))
```

Reply:
fixed